qcodes/tests/dataset/test_guid_helpers.py

```diff
@@ -77,3 +81,39 @@ def test_many_guids_from_list_str() -> None:
         'aaaaaaaa-0d00-000d-0000-017662b2a878',
         'aaaaaaaa-0d00-000d-0000-01766827cfaf']
     assert guids_from_list_str(str(guids)) == tuple(guids)
+
+
+def test_get_guid_from_multiple_run_ids(tmp_path: Path) -> None:
+    def generate_local_exp(dbpath: Path) -> List[str]:
+        with initialised_database_at(str(dbpath)):
+            guids = []
+            exp = load_or_create_experiment(experiment_name="test_guid")
+
+            p1 = Parameter('Voltage', set_cmd=None)
+            p2 = Parameter('Current', get_cmd=np.random.randn)
+
+            meas = Measurement(exp=exp)
+            meas.register_parameter(p1).register_parameter(p2, setpoints=[p1])
+
+            # Meaure for 2 times to get 2 run ids and 2 guids
+            for run in range(2):
+                with meas.run() as datasaver:
+                    for v in np.linspace(0*run, 2*run, 50):
+                        p1(v)
+                        datasaver.add_result((p1, cast(float, p1())),
+                                             (p2, cast(float, p2())))
+                guid = datasaver.dataset.guid
+                guids.append(guid)
+            datasaver.flush_data_to_database(block=True)
```

Comment:
this is not needed because it happens already upon exit of the `with meas.run() as datasaver` context

Reply:
OK. I will check it.